python/ray/air/_internal/accelerator_utils/utils.py

```diff
@@ -0,0 +1,70 @@
+import glob
+from typing import Optional, Set
+import logging
+
+import ray
+from ray._private.accelerators.hpu import HPU_PACKAGE_AVAILABLE
+from ray._private.accelerators.npu import NPU_TORCH_PACKAGE_AVAILABLE
+from ray.air._internal.accelerator_utils.device_manager import TorchDeviceManager
+from ray.air._internal.accelerator_utils.nvidia_gpu import CUDATorchDeviceManager
+from ray.air._internal.accelerator_utils.npu import NPUTorchDeviceManager
+from ray.air._internal.accelerator_utils.hpu import HPUTorchDeviceManager
+
+
+logger = logging.getLogger(__name__)
+
+
+ACCELERATOR_TORCH_DEVICE_MANAGER = {
+    "GPU": CUDATorchDeviceManager,
+    "HPU": HPUTorchDeviceManager,
+    "NPU": NPUTorchDeviceManager,
+}
+
+
+def try_import_accelerator_torch_library() -> None:
+    try:
+        if NPU_TORCH_PACKAGE_AVAILABLE:
+            # Do not register the torch.npu module even if torch_npu is installed
+            # when no npu device is detected.
+            if not len(glob.glob("/dev/davinci?")):
+                logger.warning(
+                    "It seems that there is no Ascend NPU device in "
+                    "/dev path and torch_npu will not be imported."
+                )
+            else:
+                import torch_npu  # noqa: F401
+
+        if HPU_PACKAGE_AVAILABLE:
+            import habana_frameworks.torch.hpu as torch_hpu  # noqa: F401
+
+    except ImportError:
+        raise ImportError("Could not import PyTorch")
+
+
+def get_all_torch_device_manager() -> Set[TorchDeviceManager]:
+    """Get all device manager supported by Ray"""
+    return set(ACCELERATOR_TORCH_DEVICE_MANAGER.values())
+
+
+def get_torch_device_manager_by_runtime_context() -> Optional[TorchDeviceManager]:
```

Comment:
It make sense. However, the `ScalingConfig` parameter is only passed down to the `BackendExecutor` and not to the `Backend` (and its subclass `_TorchBackend`).  We suspect this is because the `Backend` design does not actually care about the resource type. So would you suggest making the `Backend` aware of the devices, in order to explicitly determine the `DeviceManager`?

Reply:
Yes exactly. We should let the backend also knows about the accelerator type specified by the user, instead of dynamically check the available resources at runtime.

Reply:
OK, added the `resources_per_worker` field to the `Backend`. DeviceManager will be explicitly initialized when `on_training()` is called in `_TorchBackend` now.